File: app/src/main/java/com/example/amazonprimeclone/fragments/search/SearchFragmentViewModel.java
package com.example.amazonprimeclone.fragments.search;

import static com.example.amazonprimeclone.data.remote.Constants.API_KEY;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.amazonprimeclone.data.remote.Constants;
import com.example.amazonprimeclone.modal.MovieResponseResults;
import com.example.amazonprimeclone.modal.SearchMovie;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SearchFragmentViewModel extends ViewModel {

    private final MutableLiveData<List<MovieResponseResults>> movieResults;

    public SearchFragmentViewModel(){
        movieResults = new MutableLiveData<>();
    }

    public LiveData<List<MovieResponseResults>> getMovieResults(String query) {
        loadMovieResults(query);
        return movieResults;
    }

    private void loadMovieResults(String query) {
        Call<SearchMovie> movieResponseCall = Constants.retrofitService.getMoviesByQuery(API_KEY,query);
        movieResponseCall.enqueue(new Callback<SearchMovie>() {
            @Override
            public void onResponse(@NonNull Call<SearchMovie> call, @NonNull Response<SearchMovie> response) {
                SearchMovie searchMovie = response.body();
                //add to list
                if(searchMovie!=null){
                    movieResults.setValue(searchMovie.getResults());
                }

            }

            @Override
            public void onFailure(@NonNull Call<SearchMovie> call, @NonNull Throwable t) {

            }
        });
    }


}

File: app/src/main/java/com/example/amazonprimeclone/fragments/search/SearchFragmentViewModel.kt
package com.example.amazonprimeclone.fragments.search

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.amazonprimeclone.data.remote.Constants
import com.example.amazonprimeclone.modal.MovieResponseResults
import com.example.amazonprimeclone.retrofit.repository.RetrofitMovieRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class SearchFragmentViewModel  @Inject constructor(private val retrofitMovieRepository: RetrofitMovieRepository): ViewModel() {

    private val movieResults: MutableLiveData<List<MovieResponseResults>> = MutableLiveData()

    fun getMovieResults(query: String): LiveData<List<MovieResponseResults>> {
        loadMovieResults(query)
        return movieResults
    }

    private fun loadMovieResults(query: String) = viewModelScope.launch(Dispatchers.IO){
        val response = retrofitMovieRepository.getMoviesByQuery(Constants.API_KEY,query)
        if(response!=null){
            movieResults.postValue(response.results)
        }
    }
}